docs/advanced-features/react-18/server-components.md

```diff
@@ -0,0 +1,124 @@
+
+### Server Components Convention
+
+To run a component on the server, append `.server.js` to the end of the filename. For example `./pages/home.server.js` is a Server Component.
+
+For client components, add `.client.js`. For example, `./components/avatar.client.js`.
+
+You can then import other server or client components from any server component. Note: a server component **can not** be imported by a client component. Components without "server/client" extensions will be treated as "universal component" and can be used and rendered by both sides, depending on where it is imported. For example:
+
+```jsx
+// pages/home.server.js
+
+import { Suspense } from 'react'
+
+import Profile from '../components/profile.server'
+import Content from '../components/content.client'
+
+export default function Home() {
+  return (
+    <div>
+      <h1>Welcome to React Server Components</h1>
+      <Suspense fallback={'Loading...'}>
+        <Profile />
+      </Suspense>
+      <Content />
+    </div>
+  )
+}
+```
+
+The `<Home>` and `<Profile>` components will always be server-side rendered and streamed to the client, and will not be included by the client runtime. However `<Content>` will still be hydrated on the client-side, like normal React components.
+
+To see a full example, check out [link to the demo and repository](https://github.com/vercel/next-rsc-demo).
+
+## **Supported Next.js APIs**
+
+### `next/link` and `next/image`
+
+You could use `next/link` and `next/image` like before and they will be treated as client components to keep the interaction on client side.
+
+### `next/document`
+
+If you have custom `_document` To use server components you have to change your `_document` to a functional component like below. Or if you don't have any, Next.js will provide a functional fallback `_document` component.
+
+```jsx
+// pages/_document.js
+import { Html, Head, Main, NextScript } from 'next/document'
+
+export default function Document() {
+  return (
+    <Html>
+      <Head />
+      <body>
+        <Main />
+        <NextScript />
+      </body>
+    </Html>
+  )
+}
+```
+
+### `next/app`
+
+If you're use `_app.js`, the usage is same with [Custom App](/docs/advanced-features/custom-app).
+If you're use `_app.server.js` as a server component, the signature is changed as below where it only receive the `children` prop as element. You can surly wrap any other client or server components around `children` to customize the layout of your app.
+
+```js
+// pages/_app.server.js
+export default function App({ children }) {
+  return children
+}
+```
+
+### Routing
+
+Basic routes with path and queries, dynamic routes are supported as well. If you need to access the router in server components(`.server.js`), They will receive `router` instance as a prop so that you could directly accessing them without using `useRouter()` hook.
+
+```jsx
+// pages/index.server.js
+
+export default function Index({ router }) {
```

Comment:
How would this look for TS users? Could be worth mentioning a typing to use, which would be some alternative to `NextPage` I guess. Same applies to `App` component (`ServerApp` / `ServerAppProps`?)

Reply:
Will address this part later. But for custom `_app` it would be simple since it only receive the `children` prop like a basic react component